### PR TITLE
 Move some common code used by models into birdfeeder

### PR DESCRIPTION
**Pull request by sarinrp**
## Description

Common code transferred from `parrot` repo
* `from parrot.constants import HOURS_IN_DAY` and similar
* `from parrot.helpers.datetime import is_tz_aware`
* `BaseSettingsWithConfig` settings class
* `from parrot.shared.enums.str_enum import StrEnum`
* Move `parrot/aws`, except `order_event_logger.py` - this is too specific and we don't need to share it
* `from parrot.user_data_collector.exchange_api_client.local_typing import JSONTemplate`

## Author checklist

- [x] PR has corresponding ticket, and ticket is linked to the PR
- [x] Tests written **OR** decided to omit tests **OR** tests aren't needed. Explain in description if tests are not written

## Reviewer checklist

- [ ] Ticket or PR contains enough description why these changes are needed
- [ ] Coding style matches our [Code Style Guideline](https://www.notion.so/hummingbot/Code-Style-2c3db908b42242ea9a96543c1b0826be)
- [ ] Maintainability is ok
  - Naming and in-code comments adequate
  - No wheel reinventing
  - Code is not too complex
- [ ] No new usages of deprecated patterns are introduced, see [Ongoing refactorings](https://www.notion.so/hummingbot/a301c6321bc948df832e253c0a0a7aef?v=f21954e6edba43a6b6df6f1ffeae8488)
- [ ] Changes logically looks good
  - Code is doing what it's supposed to do
  - No logical conflicts with other parts
  - Corner cases handled
  - Error handling is enough
- [ ] Code is ready for operation
  - Is there enough logging to understand what the service is doing?
  - Is the new code instrumented with metrics or APM tracing? (If applicable)
  - Does the runtime configuration properly separated from the code itself (env vars, etc)

## Before merging (author or reviewer)

- [ ] Final version is tested on Development, logs OK, metrics OK, **OR** no testing is needed

## Helpful docs

- [Pull Request Guideline](https://www.notion.so/Pull-Request-and-Code-Review-16af4014d3e94f859b7bd9112511eddb)
- [Code Style Guideline](https://www.notion.so/hummingbot/Code-Style-2c3db908b42242ea9a96543c1b0826be)
- [Ongoing refactorings](https://www.notion.so/hummingbot/a301c6321bc948df832e253c0a0a7aef?v=f21954e6edba43a6b6df6f1ffeae8488)

### CODE REVIEW

File: birdfeeder/base.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import os
```
Comment (bitphage) [inline birdfeeder/base.py]: Better rename to `birdfeeder/pydantic/base_settings.py`

**Review by bitphage — CHANGES_REQUESTED** [review]
- Need to transfer AWS tests as well

Also a side note, better to use more meaningful commit messages (not just "wip") - best to use atomic commits, one commit per new feature

**Comment by sarinrp** [comment]
> * Need to transfer AWS tests as well
> 
> Also a side note, better to use more meaningful commit messages (not just "wip") - best to use atomic commits, one commit per new feature

Yes I plan to redo commits once everything is reviewed and okay to merge.

**Review by bitphage — APPROVED** [review]
LGTM

**Pull request merged**